### PR TITLE
Add CSI step registry workflow for OpenStack

### PR DESCRIPTION
**Pull request by Fedosin**
This patch registers a new step registry workflow for running CSI tests.

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/openstack/csi/OWNERS [hunk 1]
```diff
@@ -1,4 +1,13 @@
 approvers:
```
Comment (mandre) [inline ci-operator/step-registry/openshift/e2e/openstack/csi/OWNERS:1]: Please leave `openstack-approvers` and `openstack-reviewers`. We'd like to be able to make modifications to the job, for example moving it from MOC to Vexxhost, without depending on the storage team for reviews.

**Comment by Fedosin** [comment]
Blocked by https://github.com/openshift/release/pull/14516

File: ci-operator/step-registry/openshift/e2e/openstack/csi/OWNERS (hunk 1 above)
Comment (Fedosin) [reply ci-operator/step-registry/openshift/e2e/openstack/csi/OWNERS:1]: done!

**Comment by mandre** [comment]
@Fedosin this PR needs a rebase.

File: ci-operator/step-registry/storage/conf/csi-openstack-cinder/storage-conf-csi-openstack-cinder-commands.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cd /go/src/github.com/openshift/openstack-cinder-csi-driver-operator
+cp test/e2e/manifest.yaml ${SHARED_DIR}/${TEST_CSI_DRIVER_MANIFEST}
+
+# For debugging
+echo "Using ${SHARED_DIR}/${TEST_CSI_DRIVER_MANIFEST}:"
+cat ${SHARED_DIR}/${TEST_CSI_DRIVER_MANIFEST}
```
Comment (pierreprinetti) [inline ci-operator/step-registry/storage/conf/csi-openstack-cinder/storage-conf-csi-openstack-cinder-commands.sh:11]: Will this contain any secrets?

**Comment by pierreprinetti** [comment]
Looks good! I just have a question around the debug-printf

File: ci-operator/step-registry/storage/conf/csi-openstack-cinder/storage-conf-csi-openstack-cinder-commands.sh (hunk 2 above)
Comment (mandre) [reply ci-operator/step-registry/storage/conf/csi-openstack-cinder/storage-conf-csi-openstack-cinder-commands.sh:11]: Looks like no. This is the manifest file:
https://github.com/openshift/openstack-cinder-csi-driver-operator/blob/master/test/e2e/manifest.yaml

**Comment by jsafrane** [comment]
/approve 
The changes look good, however, I don't see workflow `openshift-e2e-openstack-csi` used anywhere, I expect a subsequent PR that actually uses it...

**Comment by mandre** [comment]
> The changes look good, however, I don't see workflow `openshift-e2e-openstack-csi` used anywhere, I expect a subsequent PR that actually uses it...

Correct, we forgot to mention nothing currently calls the workflow, sorry for the confusion. This is being worked on and expect a follow up PR soon.

**Pull request merged**